src/extent.c

```diff
@@ -134,13 +134,17 @@ typedef enum {
 
 static lock_result_t
 extent_rtree_leaf_elm_try_lock(tsdn_t *tsdn, rtree_leaf_elm_t *elm,
-    extent_t **result) {
-	extent_t *extent1 = rtree_leaf_elm_extent_read(tsdn, &extents_rtree,
-	    elm, true);
+    extent_t **result, bool large_only) {
+	uintptr_t bits = rtree_leaf_elm_bits_read(tsdn, &extents_rtree, elm,
+	    false);
```

Comment:
changed to a non dependent read to enforce the barrier.

Reply:
I don't think that's the right fence; I think we'd need a seq_cst fence after our modification but before the read of the neighbor (dependent vs. non-dependent is just for the lookups being acquire-y or not).

Reply:
Ah do you mean also adding a fence on the rtree update path? In that case I would suggest doing that part in a separate change, since this diff doesn’t really affect that part, and the fence overhead on a fast-ish path will need its own benchmark run to be better measured.

Reply:
No, we just need it on the merge path (and maybe not even then, it depends on how exactly).

Reply:
Actually, thinking about it even more, this only matters if the two extents might be about to be coalesced, which would mean that they're in the same state, which means that they're protected by the same extents_t lock, right? So even if the race occurs, all it means is that we do a little extra work of following the extent, and so allowing it by remaining barrier-free should be OK.